pkg/kubectl/cmd/edit.go

```diff
@@ -345,16 +345,7 @@ func getMapperAndResult(f cmdutil.Factory, args []string, options *resource.File
 	if err != nil {
 		return nil, nil, nil, "", err
 	}
-	var mapper meta.RESTMapper
-	var typer runtime.ObjectTyper
-	switch editMode {
-	case NormalEditMode:
-		mapper, typer = f.Object()
-	case EditBeforeCreateMode:
-		mapper, typer, err = f.UnstructuredObject()
```

Comment:
This `UnstructuredObject` is the direction we want to move.  We want to move away from `Object`.  `Unstructured` deals with opaque and external types, but `Object` is very opinionated about which resources exist.  To have a a truly federated API servers, we must make everything generic, not opinionated.

@kubernetes/kubectl @fabianofranz